src/lib.rs

```diff
@@ -113,20 +126,29 @@ pub fn version_check(version: &str) -> Result<(), Error> {
     }
 }
 
-/// A helper for reading the chapters of a [`Book`] into memory.
-pub fn load_files_into_memory(
+/// A helper for reading the chapters of a [`Book`] into memory, filtering out files using
+/// the given `filter`.
+pub fn load_files_into_memory<F>(
     book: &Book,
     dest: &mut Files<String>,
-) -> Vec<FileId> {
+    filter: F,
```

Comment:
It looks like a test will need to be updated because of this change:

https://travis-ci.com/github/Michael-F-Bryan/mdbook-linkcheck/jobs/389917971#L590-L596